### PR TITLE
LPS-109716 baseline

### PR DESCRIPTION
**Pull request by gabrielwas**

### CODE REVIEW

**Comment by gabrielwas** [comment]
ci:test:sf

**Comment by gabrielwas** [comment]
ci:stop

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/settings/DDMFormBuilderSettingsRetrieverHelper.java [hunk 1]
```diff
@@ -151,9 +151,8 @@ public String getSerializedDDMExpressionFunctionsMetadata(Locale locale) {
 	public String getSerializedDDMFormRules(DDMForm ddmForm) {
 		JSONSerializer jsonSerializer = _jsonFactory.createJSONSerializer();
 
-		List<SPIDDMFormRule> spiDDMFormRules =
-			_ddmFormRuleToDDMFormRuleConverter.convert(
-				ddmForm.getDDMFormRules());
+		List<SPIDDMFormRule> spiDDMFormRules = _spiDDMFormRuleConverter.convert(
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/settings/DDMFormBuilderSettingsRetrieverHelper.java]: No need for variable

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormRuleJSONDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+	private static DDMFormRule _getDDMFormRule(JSONObject jsonObject) {
+		String condition = jsonObject.getString("condition");
+
+		List<String> actions = _getDDMFormRuleActions(
+			jsonObject.getJSONArray("actions"));
+
+		DDMFormRule ddmFormRule = new DDMFormRule(condition, actions);
+
+		boolean enabled = jsonObject.getBoolean("enabled", true);
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormRuleJSONDeserializer.java]: No need for variable

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-spi/src/main/java/com/liferay/dynamic/data/mapping/spi/converter/SPIDDMFormRuleConverter.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * @author Gabriel Albuquerque
+ */
+public interface SPIDDMFormRuleConverter {
+
+	public List<SPIDDMFormRule> convert(List<DDMFormRule> ddmFormRules);
+
+	public List<DDMFormRule> convert(
+		List<SPIDDMFormRule> ddmFormRules,
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-spi/src/main/java/com/liferay/dynamic/data/mapping/spi/converter/SPIDDMFormRuleConverter.java]: Maybe like this?
```suggestion
		List<SPIDDMFormRule> spiDDMFormRules,
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormLayoutJSONSerializer.java [hunk 4]
```diff
@@ -145,6 +147,17 @@ protected void addRows(
 		jsonObject.put("rows", jsonArray);
 	}
 
+	protected void addRules(
+		JSONObject jsonObject, List<DDMFormRule> ddmFormRules) {
+
+		if (ddmFormRules.isEmpty()) {
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormLayoutJSONSerializer.java]: Change to

```suggestion
		if (ListUtil.isEmpty(ddmFormRules.isEmpty())) {
```

File: modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/dto/v2_0/util/DataLayoutUtil.java [hunk 5]
```diff
@@ -191,6 +236,55 @@ private static DataLayoutRow _toDataLayoutRow(
...
+						JSONFactoryUtil.looseSerializeDeep(
+							spiDDMFormRuleCondition),
+						Map.class)
+				).toArray(
+					Map[]::new
+				));
+
+			dataRule.setLogicalOperator(spiDDMFormRule.getLogicalOperator());
+
+			dataRules = ArrayUtil.append(dataRules, dataRule);
```
Comment (jeyvison) [inline modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/dto/v2_0/util/DataLayoutUtil.java]: why append and not just

```suggestion
			dataRules.add(dataRule);
```

**Pull request closed without merging**